### PR TITLE
#167313354 Write end-to-end tests for the application

### PR DESCRIPTION
**Pull request by nkalyesubula**
#### What does this PR do?

- This PR intends to add configurations for end-to-end and unit tests using cypress and jest frameworks

#### Description of Task to be completed?

- Write end-to-end tests for the application

#### How should this be manually tested?

- Clone the repo, and then run **npm test:cypress** or **npm test** for **end-to-end tests** or **unit tests** repectively

#### What are the relevant pivotal tracker stories?
[#167313354](https://www.pivotaltracker.com/story/show/167313354)
#### Screenshots (if appropriate)
End-to-End testing screenshot
image

unit-testing screenshot
image

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
+// This example commands.js shows you how to
+// create various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
```
Comment (nshutijonathan) [inline cypress/support/commands.js:7]: I was wondering why you push commented codes but got to know that they are generated

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by habinezadalvan — APPROVED** [review]
NICE WORK @nkalyesubula

**Review by salviosage — COMMENTED** [review]
please rebase

**Review by minega25 — APPROVED** [review]
LGTM

File: cypress/support/commands.js (hunk 1 above)
Comment (nkalyesubula) [reply cypress/support/commands.js:7]: This file is automatically generated by cypress framework

**Review by salviosage — CHANGES_REQUESTED** [review]
@nkalyesubula  great work so far, I see the application is failing on heroku please take a look on that

File: package.json [hunk 2]
```diff
@@ -22,13 +26,19 @@
...
     "eslint": "^6.4.0",
     "eslint-config-airbnb": "^18.0.1",
     "eslint-plugin-flowtype": "^3.13.0",
     "eslint-plugin-import": "^2.18.2",
     "eslint-plugin-jsx-a11y": "^6.2.3",
     "eslint-plugin-react": "^7.14.3",
-    "eslint-plugin-react-hooks": "^1.7.0"
+    "eslint-plugin-react-hooks": "^1.7.0",
+    "jest": "^24.9.0",
+    "node-sass": "^4.12.0",
```
Comment (SokoPaulSokool) [inline package.json]: node-sass is supposed to be in dependencies

File: package.json [hunk 3]
```diff
@@ -22,13 +26,19 @@
...
+    "react-test-renderer": "^16.9.0",
+    "sass-loader": "^8.0.0",
```
Comment (SokoPaulSokool) [inline package.json]: sass-loader is also supposed to be in dependencies

**Review by SokoPaulSokool — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (nkalyesubula) [reply package.json]: Fixed

File: package.json (hunk 3 above)
Comment (nkalyesubula) [reply package.json]: Fixed

**Review by Anguandia — APPROVED** [review]
LGTM, greate work @nkalyesubula

**Review by SokoPaulSokool — APPROVED** [review]
Good job

**Pull request merged**